PriceSizeApp.py

```diff
@@ -1,22 +1,25 @@
-import pandas as pd
 import numpy as np
-import math
-import datetime
-import time
-import dateutil.relativedelta
+import pandas as pd
 import os
 import fire
 
+from AlphaVantageProvider import AlphaVantageProvider
+from BarChartProvider import BarChartProvider
+
 class PriceSizeApp:
     def __init__(self,inputf):
-        self.API_KEY = 'T133DFH1BDEC1EPU'  # replace demo with your api key
-        self.BASE_URL = 'https://www.alphavantage.co/query?function=TIME_SERIES_INTRADAY_EXTENDED'
-        self.BASE_URL_Rec = 'https://www.alphavantage.co/query?function=TIME_SERIES_INTRADAY'
+        self.ALPHAVANTAGE_API_KEY = 'T133DFH1BDEC1EPU'  # replace demo with your api key
+        self.BARCHART_API_KEY = "9bff8ed715c16109a1ce5c63341bb860"
         self.dirP = "Results/"
         self.create_dirs()
         self.s_list = self.read_symbols(inputf)
         self.df_summary = self.create_summary_df()
 
+        self.period = 90
+        self.interval = "daily"
+        self.data_provider = AlphaVantageProvider(self.ALPHAVANTAGE_API_KEY)
```

Comment:
I'm confused here, how are you choosing between the AlphaVantageProvider and the BarCharProvider? It seems you just overwrite 'self.data_provider' here.